### PR TITLE
DDF-1864 Added Sample Stanag 4559 Server

### PR DESCRIPTION
**Pull request by bdeining**
DDF-1880 Added Stanag 4559 Source and Filter Delegate
DDF-1891 Added Stanag 4559 Input Transformer Stubs and Preliminaries

@kcwire 
@jrnorth 
@dcruver 
@jaymcnallie 
@shane-voisard 
@jlcsmith 
@stustison

### CODE REVIEW

**Comment by jaymcnallie** [comment]
@troymohl - you're probably interested too.

**Comment by bdeining** [comment]
https://github.com/codice/alliance/pull/3 is built on top of this PR.  Recommend not merging this, and merging that PR instead.

File: catalog/nato-app/pom.xml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+ **/
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>alliance.catalog</groupId>
+        <artifactId>catalog</artifactId>
+        <version>1.0.0-SNAPSHOT</version>
```
Comment (jlcsmith) [inline catalog/nato-app/pom.xml]: we should probably have a release plan for this project.  Usually the first release wouldn't be 1.0.

File: catalog/nato-app/src/main/resources/features.xml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          xsi:schemaLocation="http://karaf.apache.org/xmlns/features/v1.3.0 http://karaf.apache.org/xmlns/features/v1.3.0">
+
+    <feature name="catalog-stanag4559-source" install="manual" version="${project.version}"
+             description="DDF Catalog STANAG 4559 Source">
+        <bundle>mvn:alliance.catalog.nato.stanag4559/catalog-stanag4559-source/${project.version}</bundle>
+    </feature>
+
+    <feature name="nato-app" install="auto" version="${project.version}"
+             description="The DDF Nato App ::DDF Nato">
+        <feature>catalog-stanag4559-source</feature>
```
Comment (jlcsmith) [inline catalog/nato-app/src/main/resources/features.xml]: shouldn't this have a pre-req on the catalog-app?

File: catalog/nato/stanag4559/catalog-stanag4559-common/src/main/java/alliance/catalog/nato/stanag4559/common/Stanag4559CommonUtils.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+import org.jgrapht.Graph;
+import org.jgrapht.traverse.DepthFirstIterator;
+
+import alliance.catalog.nato.stanag4559.common.UCO.Edge;
+import alliance.catalog.nato.stanag4559.common.UCO.Node;
+
+public class Stanag4559CommonUtils {
+    public static Node[] getNodeArrayFromGraph(Graph<Node, Edge> graph) {
+        Object[] vertexSet = graph.vertexSet()
```
Comment (jlcsmith) [inline catalog/nato/stanag4559/catalog-stanag4559-common/src/main/java/alliance/catalog/nato/stanag4559/common/Stanag4559CommonUtils.java]: Protect against the case where graph is null?

File: catalog/nato/stanag4559/catalog-stanag4559-common/src/main/java/alliance/catalog/nato/stanag4559/common/Stanag4559MetacardType.java [hunk 4]
```diff
@@ -0,0 +1,644 @@
...
+import java.util.HashSet;
+import java.util.Set;
+
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.AttributeType;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.data.impl.AttributeDescriptorImpl;
+import ddf.catalog.data.impl.BasicTypes;
+
+public abstract class Stanag4559MetacardType implements MetacardType {
```
Comment (jlcsmith) [inline catalog/nato/stanag4559/catalog-stanag4559-common/src/main/java/alliance/catalog/nato/stanag4559/common/Stanag4559MetacardType.java]: might want to extend from MetacardTypeImpl

File: catalog/nato/stanag4559/catalog-stanag4559-common/src/main/java/alliance/catalog/nato/stanag4559/common/Stanag4559CommonUtils.java (hunk 3 above)
Comment (troymohl) [inline catalog/nato/stanag4559/catalog-stanag4559-common/src/main/java/alliance/catalog/nato/stanag4559/common/Stanag4559CommonUtils.java]: I added checks for the graph being null in each of these methods.

File: catalog/nato/stanag4559/catalog-stanag4559-common/src/main/java/alliance/catalog/nato/stanag4559/common/Stanag4559MetacardType.java (hunk 4 above)
Comment (troymohl) [inline catalog/nato/stanag4559/catalog-stanag4559-common/src/main/java/alliance/catalog/nato/stanag4559/common/Stanag4559MetacardType.java]: I didn't extend from the Impl as I didn't want the code to have to construct and pass the list of attributedescriptors and name when setting the type. I did override the default hashCode and equals.

File: catalog/nato-app/src/main/resources/features.xml (hunk 2 above)
Comment (bdeining) [inline catalog/nato-app/src/main/resources/features.xml]: Added the pre-req for `catalog-app` in https://github.com/codice/alliance/pull/3

File: catalog/nato-app/pom.xml (hunk 1 above)
Comment (bdeining) [inline catalog/nato-app/pom.xml]: Moved to `0.1-SNAPSHOT` in https://github.com/codice/alliance/pull/3

File: catalog/nato/stanag4559/catalog-stanag4559-common/src/main/java/alliance/catalog/nato/stanag4559/common/Stanag4559VideoMetacardType.java [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+
+    /**
+     * The standard rate (in frames per second (FPS)) at which the motion imagery was captured.
+     */
+    public static final String FRAME_RATE = "frameRate";
+
+    /**
+     * The frame vertical resolution.
+     */
+    public static final String NUM_ROWS = "numRows";
```
Comment (jlcsmith) [inline catalog/nato/stanag4559/catalog-stanag4559-common/src/main/java/alliance/catalog/nato/stanag4559/common/Stanag4559VideoMetacardType.java]: why strings for these fields?  We should use the proper numeric type instead

Comment (jlcsmith) [inline catalog/nato/stanag4559/catalog-stanag4559-common/src/main/java/alliance/catalog/nato/stanag4559/common/Stanag4559VideoMetacardType.java]: that is probably a global comment for all the MetacardType classes

Comment (jlcsmith) [inline catalog/nato/stanag4559/catalog-stanag4559-common/src/main/java/alliance/catalog/nato/stanag4559/common/Stanag4559VideoMetacardType.java]: that is probably a global comment for all the MetacardType classes

Comment (jlcsmith) [inline catalog/nato/stanag4559/catalog-stanag4559-common/src/main/java/alliance/catalog/nato/stanag4559/common/Stanag4559VideoMetacardType.java]: disregard I'm dumb

**Comment by bdeining** [comment]
https://github.com/codice/alliance/pull/5.  Closing this PR and breaking up some of the code to make it easier to review.  All comments were addressed.

**Pull request closed without merging**